components/TheNavBar.vue

```diff
@@ -0,0 +1,137 @@
+<template>
+  <nav class="bg-white w-full shadow-md fixed flex items-center z-50">
+    <div class="container mx-auto px-4 flex justify-between">
+      <ul id="menu_list list-none" v-bind:class="{ active: toggle == true }">
+        <li><a class="text-lg font-bold no-underline" href="#about">Despre Alpha</a></li>
+        <li><a class="text-lg font-bold no-underline" href="#testimonials">Testimoniale</a></li>
+        <li><a class="text-lg font-bold no-underline" href="#register">Încriere</a></li>
+      </ul>
+      <button class="toggle" @click="toggle = !toggle">
+        <span v-bind:class="{ 'translate-y-1/2 rotate-45': toggle == true }" class="nav_btn_activ absolute bg-black w-9 h-1 rounded-sm transform translate-y-1.5 top-1/2"></span>
+        <span v-bind:class="{ 'translate-y-1/2 -rotate-45': toggle == true }" class="nav_btn_activ absolute bg-black w-9 h-1 rounded-sm transform -translate-y-1.5 top-1/2"></span>
+      </button>
+      <a href="#top-section"><img class="logo" src="../static/logo.svg" alt=""></a>
+    </div>
+  </nav>
+</template>
+
+<script>
+export default {
+  data () {
+    return {
+      toggle: false
+    }
+  },
+  methods: {
+    handleScroll (event) {
+      if (window.scrollY > 0) {
+        document.querySelector('nav').classList.add('nav-on-scroll')
+        document.querySelector('.logo').classList.add('logo-on-scroll')
+      } else {
+        document.querySelector('nav').classList.remove('nav-on-scroll')
+        document.querySelector('.logo').classList.remove('logo-on-scroll')
+      }
+    }
+  },
```

Comment:
we should get rid of this, classes should update based on reactive data. here's what you should do:

1. add a `hasScrolled: false` data property (right under line 22)
2. the only thing you should change in the `if-else` block is `this.hasScrolled`.
3. on the `nav` and `.logo` elements, **add** a `:class` binding (along with the `class` attribute, just trust me for now). Here's an example for the `nav` element: `:class="{ 'nav-on-scroll': hasScrolled }"`. Apply the same reasoning to the other element.